network-api/networkapi/wagtailpages/tests/buyersguide/test_article_page.py

```diff
@@ -206,3 +210,107 @@ def test_secondary_related_articles_no_related_articles(self):
         result = article_page.get_secondary_related_articles()
 
         self.assertListEqual(result, [])
+
+    @staticmethod
+    def generate_form_data(article_page, data: dict) -> dict:
+        """
+        Generate valid form data for the BuyersGuide article page form.
+
+        We add the extra fields below to the dict as all these fields are set as
+        required, and the form will return a validation error against them if left out.
+
+        Since we have the "body" field set as required, we need to submit at least
+        one StreamField in the form, or else the form will return a validation error.
+        """
+        return nested_form_data(
+            {
+                **data,
+                "title": article_page.title,
+                "slug": article_page.slug,
```

Comment:
We don't want to test anything about the `title` and `slug` fields coming from some existing article page. Have you considered setting these as strings directly? 

In that case, we would also not need to create an article page in each test.

Reply:
Hi Tibor, thanks for the suggested. I think this makes sense because this would simplify the test further, and make sure that this function produces the same data for every test. 

I have implemented this in the latest commit!